src/Pages/Reconstruction.jsx

```diff
@@ -110,21 +159,70 @@ export class Reconstruction extends Component {
               <br />
               <input
                 className='form-control-file'
-                id='image-upload'
                 name='images'
                 type='file'
-                onChange={this.onFileChange}
                 multiple
+                accept='image/png, image/jpeg'
+                required
               />
               <small className='form-text text-muted'>
                 Select multiple files at once.
               </small>
             </div>
 
-            <button type='submit' className='btn btn-primary my-2'>
-              Start Reconstruction
+            <button
+              disabled={this.state.uploading && true}
+              type='submit'
+              className='btn btn-primary my-2'
+            >
+              Upload
             </button>
           </form>
+
+          {/* uploading spinner */}
+          {this.state.uploading && (
+            <span>
+              <FontAwesomeIcon icon='spinner' pulse className='mx-5' />
+              Uploading...
+            </span>
+          )}
+
+          {/* Size error message */}
+          {this.state.uploadFail && (
+            <span className='alert alert-danger'>{IMAGE_SIZE_ERROR}</span>
+          )}
+
+          {/* Success Message */}
+          {this.state.uploadSuccess && (
+            <div>
+              {' '}
+              <span className='alert alert-success'>
+                {IMAGE_UPLOAD_SUCCESS}
+              </span>
+              <button
+                className='btn btn-success mx-2'
+                onClick={this.clickHandler}
+              >
+                <Link to={RECONSTRUCTION_URL}>Reconstruct Again</Link>
```

Comment:
I don't think we need to redirect the user.
```suggestion
                Reconstruct Again
```